index.js

```diff
@@ -86,72 +84,57 @@ Collector.prototype._notifyIfFlushed = function(callback, err) {
     setImmediate(callback, err);
 };
 
-Collector.prototype._flushSeries = function(seriesName, points, callback) {
+Collector.prototype._flushPoints = function(points, callback) {
     if (!points || points.length === 0) {
         return;
     }
     var self = this;
 
     // only send N points at a time to avoid making requests too large
-    var spliceIndex;
-    if (self._client.protocol == 'udp:' || self._client.protocol == 'line') {
-      spliceIndex = self.computePointCountToSend(points.map(function (point) {
-        return JSON.stringify(point).length;
-      }), MTU_SIZE);
-    } else {
-      spliceIndex = 50;
-    }
+    var spliceIndex = 50
     var batch = points.slice(0, spliceIndex);
     points = points.slice(spliceIndex);
     var opt = { precision: self._time_precision };
 
     self._flushesInFlight++;
-    self._client.writePoints(seriesName, batch, opt, function(err) {
+    self._client.writePoints(batch, opt)
+      .catch(function(err) {
+        self._flushesInFlight--;
```

Comment:
If you want to replicate previous behavior, consider supplying both `onFulfilled` and `onRejected` instead of `catch` and `then` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then)